packages/utils/lib/utils/scheduleUtils.ts

```diff
@@ -180,30 +181,20 @@ export function getTimezone(schedule: Location | Practitioner | HealthcareServic
   return timezone;
 }
 
-export const getAppointmentTimezone = async (
-  appointment: Appointment,
-  relevantResources: Resource[]
-): Promise<string> => {
-  const resourceReference = appointment.participant?.find(
-    (participant) =>
-      participant.actor?.reference?.startsWith('Location/') ||
-      participant.actor?.reference?.startsWith('HealthcareService/') ||
-      participant.actor?.reference?.startsWith('Practitioner/')
-  )?.actor?.reference;
-
-  if (!resourceReference) {
-    console.log('no valid resource reference found in appointment', appointment.id);
-  }
+export const getAppointmentTimezone = (appointment: Appointment, scheduleResources: Resource[]): string => {
+  const participantReferences =
+    appointment.participant?.map((participant) => participant.actor?.reference).filter(Boolean) ?? [];
 
-  const resource = relevantResources.find(
-    (res) => res.resourceType && res.id && `${res.resourceType}/${res.id}` === resourceReference
+  const scheduleResource = scheduleResources.find(
+    (resource) =>
+      resource.resourceType && resource.id && participantReferences.includes(`${resource.resourceType}/${resource.id}`)
   );
 
-  if (!resource) {
-    console.log('referenced resource not found in provided resources', resourceReference);
+  if (!scheduleResource) {
+    console.log('referenced resource not found in provided resources', scheduleResources);
```

Comment:
Maybe smething like this?
```suggestion
    console.error('Resource with Schedule extension not found in appointment.participant resources', scheduleResources);
```